### PR TITLE
feat(pt): add model branch alias

### PR DESCRIPTION
**Pull request by iProzd**
Introduces model branch alias and info fields to model configuration, adds utility functions for handling model branch dictionaries, and updates related modules to use alias-based lookup and provide detailed branch information. Enhances multi-task model usability and improves logging of available model branches.

example:
```
dp --pt show 0415_compat_new.pt model-branch

[2025-08-14 10:05:54,246] DEEPMD WARNING To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, DP_INTRA_OP_PARALLELISM_THREADS, and DP_INTER_OP_PARALLELISM_THREADS. See https://deepmd.rtfd.io/parallelism/ for more information.
[2025-08-14 10:05:59,122] DEEPMD INFO    This is a multitask model
[2025-08-14 10:05:59,122] DEEPMD INFO    Available model branches are ['Dai2023Alloy', 'Zhang2023Cathode', 'Gong2023Cluster', 'Yang2023ab', 'UniPero', 'Huang2021Deep-PBE', 'Liu2024Machine', 'Zhang2021Phase', 'Jinag2021Accurate', 'Chen2023Modeling', 'Wen2021Specialising', 'Wang2022Classical', 'Wang2022Tungsten', 'Wu2021Deep', 'Huang2021Deep-PBEsol', 'Transition1x', 'Wang2021Generalizable', 'Wu2021Accurate', 'MPTraj', 'Li2025APEX', 'Shi2024SSE', 'Tuo2023Hybrid', 'Unke2019PhysNet', 'Shi2024Electrolyte', 'ODAC23', 'Alex2D', 'OMAT24', 'SPICE2', 'OC20M', 'OC22', 'Li2025General', 'RANDOM'], where 'RANDOM' means using a randomly initialized fitting net.
[2025-08-14 10:05:59,125] DEEPMD INFO    Detailed information:
+-----------------------+------------------------------+--------------------------------+--------------------------------+
| Model Branch          | Alias                        | description                    | observed_type                  |
+-----------------------+------------------------------+--------------------------------+--------------------------------+
| Dai2023Alloy          | Alloys, Domains_Alloy        | The dataset contains           | ['La', 'Fe', 'Ho', 'Cu', 'Sn', |
|                       |                              | structure-energy-force-virial  | 'Cd', 'Y', 'Be', 'V', 'Sm',    |
|                       |                              | data for 53 typical metallic   | 'In', 'Pr', 'Mo', 'Mn', 'Gd',  |
|                       |                              | elements in alloy systems,     | 'Ru', 'Nd', 'Li', 'Tm', 'K',   |
|                       |                              | including ~9000 intermetallic  | 'Pt', 'Ir', 'Na', 'Hf', 'Dy',  |
|                       |                              | compounds and FCC, BCC, HCP    | 'Ca', 'Nb', 'Au', 'Sr', 'Si',  |
|                       |                              | structures. It consists of two | 'Ge', 'Co', 'W', 'Cr', 'Zn',   |
|                       |                              | parts: DFT-generated relaxed   | 'Ag', 'Ti', 'Ni', 'Zr', 'Pd',  |
|                       |                              | and deformed structures, and   | 'Os', 'Ta', 'Rh', 'Sc', 'Tb',  |
|                       |                              | randomly distorted structures  | 'Al', 'Ga', 'Re', 'Lu', 'Er',  |
|                       |                              | produced covering pure metals, | 'Mg', 'Ce', 'Pb']              |
|                       |                              | solid solutions, and           |                                |
|                       |                              | intermetallics with vacancies. |                                |
+-----------------------+------------------------------+--------------------------------+--------------------------------+
| OMAT24                | Default, Materials, Omat24   | OMat24 is a large-scale open   | ['La', 'Fe', 'Cu', 'Cd', 'Be', |
|                       |                              | dataset containing over 110    | 'Ar', 'V', 'Sm', 'In', 'Pm',   |
|                       |                              | million DFT calculations       | 'Pr', 'Mn', 'Ru', 'He', 'Nd',  |
|                       |                              | spanning diverse structures    | 'Th', 'Pa', 'K', 'Pt', 'Yb',   |
|                       |                              | and compositions. It is        | 'Dy', 'Sr', 'Co', 'Np', 'Cr',  |
|                       |                              | designed to support AI-driven  | 'Tl', 'Br', 'Se', 'Ni', 'Zr',  |
|                       |                              | materials discovery by         | 'Pu', 'O', 'Xe', 'Tb', 'Ga',   |
|                       |                              | providing broad and deep       | 'Lu', 'H', 'Ne', 'Er', 'Ce',   |
|                       |                              | coverage of chemical space.    | 'I', 'Kr', 'Ho', 'Cs', 'Sn',   |
|                       |                              |                                | 'Rb', 'Y', 'N', 'F', 'Mo',     |
|                       |                              |                                | 'Gd', 'B', 'Li', 'Tm', 'Sb',   |
|                       |                              |                                | 'Ir', 'Hf', 'Na', 'Ca', 'Nb',  |
|                       |                              |                                | 'Au', 'As', 'Si', 'Ge', 'W',   |
|                       |                              |                                | 'Zn', 'Hg', 'Ag', 'Bi', 'Ti',  |
|                       |                              |                                | 'Os', 'Cl', 'Pd', 'P', 'U',    |
|                       |                              |                                | 'Tc', 'Ta', 'Ba', 'Rh', 'Sc',  |
|                       |                              |                                | 'C', 'S', 'Te', 'Al', 'Re',    |
|                       |                              |                                | 'Eu', 'Mg', 'Pb', 'Ac']        |
+-----------------------+------------------------------+--------------------------------+--------------------------------+
```

## Summary by CodeRabbit

* **New Features**
  * Alias-based multi-task branch selection for evaluation and fine-tuning; new API to query model alias/branch info; show now prints a detailed model-branch table.

* **Documentation**
  * Model config gains optional fields to declare branch aliases and per-branch info (PyTorch-only).

* **Examples**
  * Added a two-task PyTorch example demonstrating aliases, shared components, and per-branch info.

* **Tests**
  * Tests include the new example and now filter out table-like show output.

### CODE REVIEW

File: deepmd/utils/argcheck.py [hunk 1]
```diff
@@ -2320,6 +2324,20 @@ def standard_model_args() -> Argument:
                 [fitting_variant_type_args()],
                 doc=doc_fitting,
             ),
+            Argument(
+                "model_branch_alias",
+                list,
```
Comment (njzjz) [inline deepmd/utils/argcheck.py]:
```suggestion
                list[str],
```

**Review by njzjz — CHANGES_REQUESTED** [review]

File: deepmd/pt/infer/deep_eval.py [hunk 2]
```diff
@@ -253,6 +266,17 @@ def get_has_hessian(self):
         """Check if the model has hessian."""
         return self._has_hessian
 
+    def get_model_branch(self):
```
Comment (wanghan-iapcm) [inline deepmd/pt/infer/deep_eval.py:283]: Is this method used anywhere?

File: deepmd/entrypoints/show.py [hunk 3]
```diff
@@ -33,10 +37,15 @@ def show(
             )
         model_branches = list(model_params["model_dict"].keys())
         model_branches += ["RANDOM"]
+        model_alias_dict, model_branch_dict = get_model_dict(model_params["model_dict"])
```
Comment (wanghan-iapcm) [inline deepmd/entrypoints/show.py]:
```suggestion
        _, model_branch_dict = get_model_dict(model_params["model_dict"])
```

File: deepmd/pt/infer/deep_eval.py [hunk 4]
```diff
@@ -116,15 +119,25 @@ def __init__(
             self.model_def_script = self.input_param
             self.multi_task = "model_dict" in self.input_param
             if self.multi_task:
+                model_alias_dict, model_branch_dict = get_model_dict(
+                    self.input_param["model_dict"]
+                )
                 model_keys = list(self.input_param["model_dict"].keys())
+                if head is None and "Default" in model_alias_dict:
+                    head = "Default"
```
Comment (Chengqian-Zhang) [inline deepmd/pt/infer/deep_eval.py:130]: Do we need to print the information of the model branch user uses especially user does not specify the model branch in the command line?

File: deepmd/pt/infer/deep_eval.py (hunk 2 above)
Comment (iProzd) [reply deepmd/pt/infer/deep_eval.py:283]: It's designed for agent access using DP calculator.

**Comment by iProzd** [comment]
> 1. should the branch names case sensitive or not ?
> 2. we may adjust first and 2nd col width of the printed table to the maximum size of the branch name and alias name.

1. For inner design, case sensitive is better, which is assumed in both state keys in checkpoint or model keys in multi-task. While for user interface, maybe it's better we first accept case sensitive names, then case-insensitive names by normalizing to lowercase and mapping to the first matching key.
2. Fixed.

**Review by anyangml — APPROVED** [review]

File: deepmd/utils/argcheck.py [hunk 5]
```diff
@@ -2305,6 +2305,10 @@ def model_args(exclude_hybrid=False):
 def standard_model_args() -> Argument:
     doc_descrpt = "The descriptor of atomic environment."
     doc_fitting = "The fitting of physical properties."
+    doc_model_branch_alias = (
+        "The alias of this model branch. This is only used in the multi-task model."
+    )
+    doc_info = "The information of this model branch. This is only used in the multi-task model."
```
Comment (njzjz) [inline deepmd/utils/argcheck.py]: After reading the documentation, I cannot get how to set these two arguments. (and I think no one knows how to set them by only reading the documentation)

Comment (njzjz) [inline deepmd/utils/argcheck.py:2311]: After reading the documentation, I cannot get how to set these two arguments. (and I think no one knows how to set them)

**Review by Chengqian-Zhang — APPROVED** [review]
